std/algorithm/searching.d

```diff
@@ -3873,6 +3873,21 @@ if (isInputRange!Range && !isInfinite!Range &&
     assert([BigInt(2), BigInt(3)].maxElement == BigInt(3));
 }
 
+// https://issues.dlang.org/show_bug.cgi?id=24596
+@safe unittest
+{
+    static class A {
+        int i;
+        int getI() @safe => i;
+        this(int i) @safe { this.i = i; }
+    }
+    auto arr = [new A(2), new A(3)];
+
+    arr.maxElement!(a => a.getI);
+
+    assert(arr[0].getI == 2);
```

Comment:
stupid question, what does this test case have to do with `std.typecons.Rebindable2`?

Reply:
`Rebindable2` is the internal helper I added to allow `extremum` to handle immutable types. A whole bunch of `std.algorithm` code, including `maxElement`, are just wrappers around `extremum`.  And this code is ~how I actually found the error.

Reply:
To elaborate, because `extremum` stores the extreme element in a `Rebindable2`, that variable is first set to `arr[0]`, then replaced with `arr[1]` - to do this, it clears `arr[0]`, which due to `destroy`'s amazing multifunctional behavior deletes `arr[0]`'s vtable pointer.